### DISCUSSION TITLE
[FEATURE REQUEST] Add the OpenSSF Scorecard GitHub Action

### DISCUSSION THREAD

**Discussion by pnacht** [saltstack/salt#67356] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
Open-source supply-chain attacks are [increasing every year][sonatype]. Beyond the infamous [SolarWinds][solarwinds] and [Codecov][codecov] attacks, there have also been multiple smaller but significant supply-chain disruptions such as [left-pad][leftpad], [colors/faker.js][faker], [coa/rc][coa] and [ua-parser-js][ua].

**Describe the solution you'd like**
I'm working on behalf of Google and the [Open Source Security Foundation][ossf] to help essential open-source projects improve their supply-chain security. Given salt's significant presence in the sensitive IT automation space, the OpenSSF has identified it as one of the 100 most critical open source projects.

Would you consider adopting an OpenSSF tool called [Scorecards][sc]? Scorecards runs dozens of automated security [checks][checks] to help maintainers better understand their project's supply-chain security posture. It is developed by the OpenSSF, [in partnership with GitHub][sc-blog].

While glancing at salt's issues and PRs, I noticed the project's already interacted with the OpenSSF and Scorecards: #34597 suggested adding the OpenSSF (then Core Infrastructure Initiative, or CII) Best Practices badge and #62317 implemented a supply-chain security improvement suggested by Scorecards.

However, the OpenSSF has also developed the [Scorecard GitHub Action][sc-gha], which aims to automate the detection of such improvements. The Action is very lightweight and runs on every change to the repository's main branch. The results of its checks are available on the project's [security dashboard](https://github.com/saltstack/salt/security), and include suggestions on how to solve any issues (see examples below). Optionally, it can also publish your results to the OpenSSF REST API, which allows a [badge][badge] with the project's score to be added to its README. This Action has been adopted by 1800+ projects already. 

Would you be interested in a PR which adds this Action? 

![Code scanning dashboard with multiple alerts, including Code-Review and Token-Permissions][img-security]

![Detail of a Token-Permissions alert, indicating the specific file and remediation steps][img-detail]

[badge]: https://openssf.org/blog/2022/09/08/show-off-your-security-score-announcing-scorecards-badges/
[coa]: https://www.bleepingcomputer.com/news/security/popular-coa-npm-library-hijacked-to-steal-user-passwords/
[codecov]: https://blog.gitguardian.com/codecov-supply-chain-breach/
[faker]: https://www.bleepingcomputer.com/news/security/dev-corrupts-npm-libs-colors-and-faker-breaking-thousands-of-apps/
[leftpad]: https://qz.com/646467/how-one-programmer-broke-the-internet-by-deleting-a-tiny-piece-of-code/
[checks]: https://github.com/ossf/scorecard#scorecard-checks
[ossf]: https://openssf.org/
[sc]: https://github.com/ossf/scorecard
[sc-blog]: https://github.blog/2022-01-19-reducing-security-risk-oss-actions-opensff-scorecards-v4/
[sc-gha]: https://github.com/ossf/scorecard-action
[solarwinds]: https://en.wikipedia.org/wiki/2020_United_States_federal_government_data_breach
[sonatype]: https://www.sonatype.com/resources/state-of-the-software-supply-chain-2021
[ua]: https://www.bleepingcomputer.com/news/security/popular-npm-library-hijacked-to-install-password-stealers-miners/

[img-security]: https://user-images.githubusercontent.com/15221358/190184391-84ca1844-259a-4b3b-9c86-74adadbea7f1.png
[img-detail]: https://user-images.githubusercontent.com/15221358/190184600-ee8d3b39-077e-416a-8711-1b5fb01cf0b3.png

**Comment by welcome** [comment]
Hi there! Welcome to the Salt Community! Thank you for making your first contribution. We have a lengthy process for issues and PRs. Someone from the Core Team will follow up as soon as possible. In the meantime, here’s some information that may help as you continue your Salt journey.
Please be sure to review our [Code of Conduct](https://github.com/saltstack/salt/blob/master/CODE_OF_CONDUCT.md). Also, check out some of our community resources including:

  - [Community Wiki](https://github.com/saltstack/community/wiki)
  - [Salt’s Contributor Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html)
  - [Join our Community Slack](https://join.slack.com/t/saltstackcommunity/shared_invite/zt-3av8jjyf-oBQ2M0vhXOhJpNpRkPWBvg)
  - [IRC on LiberaChat](https://web.libera.chat/#salt)
  - [Salt Project YouTube channel](https://www.youtube.com/channel/UCpveTIucFx9ljGelW63-BWg)
  - [Salt Project Twitch channel](https://www.twitch.tv/saltprojectoss)

There are lots of ways to get involved in our community. Every month, there are around a dozen opportunities to meet with other contributors and the Salt Core team and collaborate in real time. The best way to keep track is by subscribing to the Salt Community Events Calendar.
If you have additional questions, email us at saltproject@vmware.com. We’re glad you’ve joined our community and look forward to doing awesome things with you!
